test/server/connection_handler_test.cc

```diff
@@ -172,13 +178,15 @@ TEST_F(ConnectionHandlerTest, DestroyCloseConnections) {
 
   Network::MockListener* listener = new NiceMock<Network::MockListener>();
   Network::ListenerCallbacks* listener_callbacks;
+  TestListener* test_listener = addListener(1, true, false, "test_listener");
+  EXPECT_CALL(test_listener->socket_, socketType())
```

Comment:
add OnCall(test_listener->socket_, socketType()) into TestListener constructor instead of EXPECT_CALL everywhere. TestListener() can take an argument to indicate TCP or Udp